src/lib/storage/dictionary_segment.hpp

```diff
@@ -55,7 +55,17 @@ class DictionarySegment : public BaseDictionarySegment {
    */
   EncodingType encoding_type() const final;
 
+  // Returns the first value ID that refers to a value >= the search value and INVALID_VALUE_ID if all values are
+  // smaller than the search value. Here, INVALID_VALUE_ID does not represent NULL (which isn't stored in the
+  // dictionary anyway). Imagine a segment with values from 1 to 10. A scan for `WHERE a < 12` would retrieve
+  // `lower_bound(12) == INVALID_VALUE_ID` and compare all values in the attribute vector to `< INVALID_VALUE_ID`.
+  // Thus, returning INVALID_VALUE_ID makes comparisons much easier. However, the caller has to make sure that
+  // NULL values stored in the attribute vector (stored with a value ID of unique_values_count()) are excluded.
+  // See #1471 for a deeper discussion.
   ValueID lower_bound(const AllTypeVariant& value) const final;
```

Comment:
(Why) is it not possible to return a std::nullopt here, for "not found"? I think that would prevent half of the possible confusion around what this function returns. We have to check the return value anyway, right? So there would be no perf loss from returning a std::optional?

Reply:
I started that, but that just moves the problem to the caller. A value id that is higher than everything else is exactly what we want for `< non_existing_value`.